tests/unit/cluster/replica-migration.tcl

```diff
@@ -354,26 +349,14 @@ start_cluster 4 4 {tags {external:skip cluster} overrides {cluster-node-timeout
             fail "valkey-cli --cluster rebalance returns non-zero exit code, output below:\n$result"
         }
 
-        # Wait for server 7 role response to change.
+        # Make sure server 3 lost its replica (server 7) and server 7 becomes a replica of primary 0.
         wait_for_condition 1000 50 {
-            [R 7 role] ne $old_role_response
+            [s -3 connected_slaves] eq 0 &&
+            [get_my_primary_peer 7] eq $addr
         } else {
             puts "R 3 role: [R 3 role]"
             puts "R 7 role: [R 7 role]"
-            fail "Server 7 role response has not changed"
+            fail "Server 3 and 7 role response has not changed"
         }
-
-        wait_for_cluster_propagation
-        wait_for_cluster_state "ok"
-
-        # Make sure server 3 is still a primary and has no replicas.
-        assert_equal [s -3 role] {master}
-        assert_equal [lindex [R 3 role] 2] {}
-
-        # And server 7 becomes a replica of another primary.
-        set new_primary_ip [lindex [R 7 role] 1]
-        set new_primary_port [lindex [R 7 role] 2]
```

Comment:
We don't need to check this?

Reply:
it was covered by the wait_for_condition, they are actually do the same check

Reply:
You're right, I see it.